File: String/6. ZigZig Conversion/Longest Substring Without Repeating Character.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        
        if numRows == 1:
            return s
        
        rows = [""]*numRows
        
        cycle = 2*numRows - 2
        result = ""
        
        level = 0 #层数
        aux = -1
        for i in s:
            rows[level] += i
            if(level == 0 or level == numRows -1):
                aux *= -1
            level += aux
        
        for i in rows:
            result += i
                    
        return result  

File: String/0006. ZigZig Conversion/Longest Substring Without Repeating Character.cpp
class solution{
public:
	string convert(string s, int numRows){
		if (s.size() == 1) return s;
		string resString;
		int stringLen = s.size();
		int cycleLen = 2 * numRows - 2;

		for (int i = 0; i<numRows; i++){
			for (int j = 0; j + i<stringLen; j += cycleLen){
				resString += s[i + j];
				if (i != 0 && i != numRows - 1 && j + cycleLen - i < stringLen){
					resString += s[j + cycleLen - i];
				}
			}
			return resString;
		}
	};